### PR TITLE
DEV2-1940: Add handling of open hub with action

### PR DESCRIPTION
**Pull request by ofekby**

### CODE REVIEW

File: src/notifications/executeNotificationAction.ts [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
-  if (selectedActions?.includes(MessageActions.OPEN_NOTIFICATIONS_IN_HUB)) {
+  if (selectedActions?.includes(MessageActionsEnum.OPEN_NOTIFICATIONS_IN_HUB)) {
     return openHub(
       StateType.NOTIFICATION,
       `/home?${NOTIFICATIONS_OPEN_QUERY_PARAM}`
     )();
   }
 
+  const openHubWithActions = selectedActions
+    ?.map((action) => action as OpenHubWithAction)
```
Comment (assapir) [inline src/notifications/executeNotificationAction.ts:25]: This seems like a bug? What if the message don't have the action? Can we actually filter by something and not only make TS happy?

Comment (assapir) [reply src/notifications/executeNotificationAction.ts:25]: Make sure it has queryParams and path?

File: src/notifications/executeNotificationAction.ts [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
+    .filter((action) => action.OpenHubWith)
+    .map(({ OpenHubWith: { path, query_params } }) =>
+      openHub(StateType.NOTIFICATION, buildFullHubPath(path, query_params))()
+    );
+
+  if (openHubWithActions?.length) {
+    return Promise.all(openHubWithActions).then();
```
Comment (assapir) [inline src/notifications/executeNotificationAction.ts:32]: Why the `then`?

File: src/notifications/executeNotificationAction.ts [hunk 3]
```diff
@@ -1,23 +1,44 @@
...
+  }
+
   return Promise.resolve();
 }
+
+function buildFullHubPath(path: string, params: [string, string][]) {
+  return `${path}?${buildQueryParamsString(params)}`;
+}
+
+function buildQueryParamsString(params: [string, string][]) {
```
Comment (assapir) [inline src/notifications/executeNotificationAction.ts:42]: Use URLSearchParams?

**Review by assapir — APPROVED** [review]
If possible we should add tests

File: src/notifications/executeNotificationAction.ts (hunk 2 above)
Comment (ofekby) [reply src/notifications/executeNotificationAction.ts:32]: To change it from returning `Promise<void[]>` to `Promise<void>`

File: src/notifications/executeNotificationAction.ts (hunk 1 above)
Comment (ofekby) [reply src/notifications/executeNotificationAction.ts:25]: It is just typescript, once we have this weird mixed array you need to actually check if the property exists on the object and that the way to identify it.

File: src/notifications/executeNotificationAction.ts (hunk 3 above)
Comment (ofekby) [reply src/notifications/executeNotificationAction.ts:42]: Nice

**Review by amirbilu — APPROVED** [review]

**Pull request merged**